File: app/src/main/java/me/kr328/nevo/decorators/smscaptcha/utils/PatternUtils.java
package me.kr328.nevo.decorators.smscaptcha.utils;

import java.util.regex.Pattern;
import java.util.regex.PatternSyntaxException;

public class PatternUtils {
    @SuppressWarnings("ResultOfMethodCallIgnored")
    public static boolean checkPatternValid(String pattern) {
        try {
            Pattern.compile(pattern);
        } catch (PatternSyntaxException e) {
            return false;
        }
        return true;
    }

    public static Pattern compilePattern(String pattern, String def, int flags) {
        try {
            return Pattern.compile(pattern, flags);
        } catch (PatternSyntaxException ignored) {
        }
        return Pattern.compile(def, flags);
    }
}

File: app/src/main/java/me/kr328/nevo/decorators/smscaptcha/utils/PatternUtils.kt
package me.kr328.nevo.decorators.smscaptcha.utils

import java.util.regex.Pattern
import java.util.regex.PatternSyntaxException

object PatternUtils {
    fun checkPatternValid(pattern: String): Boolean {
        try {
            Pattern.compile(pattern)
        } catch (e: PatternSyntaxException) {
            return false
        }
        return true
    }

    fun compilePattern(pattern: String, def: String, options: Set<RegexOption>): Regex {
        try {
            return Regex(pattern, options)
        } catch (ignored: PatternSyntaxException) {
        }
        return Regex(def, options)
    }
}